src/middlewares/debugMiddleware.ts

```diff
@@ -0,0 +1,105 @@
+import { CONFIG } from '../config'
+import * as crypto from '@shardus/crypto-utils'
+import { nestedCountersInstance } from '../utils/nestedCounters'
+import { rateLimit } from 'express-rate-limit'
+import { DevSecurityLevel } from '../types'
+import { RequestHandler } from 'express'
+
+const MAX_COUNTER_BUFFER_MILLISECONDS = 10000
+let lastCounter = 0
+let multiSigLstCounter = 0
+
+const limiter = rateLimit({
+  windowMs: CONFIG.debugEndpointRateLimiting.window,
+  limit: CONFIG.debugEndpointRateLimiting.limit,
+})
+
+/** Helper Functions */
+
+function verify(obj: crypto.SignedObject, expectedPk?: string): boolean {
+  try {
+    if (expectedPk) {
+      if (obj.sign.owner !== expectedPk) return false
+    }
+    return crypto.verifyObj(obj)
+  } catch (e) {
+    console.error(`Error in verifying object ${JSON.stringify(obj)}`, e)
+    return false
+  }
+}
+
+export function ensureKeySecurity(pubKey: string, level: number): boolean {
+  const devPublicKeys = getDevPublicKeys()
+  // eslint-disable-next-line security/detect-object-injection
+  const pkClearance = devPublicKeys[pubKey]
+  return pkClearance !== undefined && pkClearance >= level
+}
+
+export function getDevPublicKeys(): { [pubkey: string]: number } {
+  return CONFIG.devPublicKeys || {}
+}
+
+// This function is used to check if the request is authorized to access the debug endpoint
+export function handleDebugAuth(_req: any, res: any, next: any, authLevel: any) {
+  try {
+    //auth with a signature
+    if (_req.query.sig != null && _req.query.sig_counter != null) {
+      const devPublicKeys = getDevPublicKeys() // This should return list of public keys
+      const requestSig = _req.query.sig
+      // Check if signature is valid for any of the public keys
+      for (const ownerPk in devPublicKeys) {
+        let sigObj = {
+          route: _req.route.path,
+          count: String(_req.query.sig_counter),
+          sign: { owner: ownerPk, sig: requestSig },
+        }
```

Comment:
signing path w/o params is risky. in this exact implementation no vulnerability cuz of counter